File: Data Structures & Algorithms/valid-binary-search-tree/submission-3.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @return {boolean}
     */
    isValid(current,left,right){
        if(!current) return true
        if(left>=current.val || current.val>=right) return false
        return (this.isValid(current.left,left,current.val) && this.isValid(current.right,current.val,right))
    }
    isValidBST(root) {
        return this.isValid(root,-1001,1001)
    }
}

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-3.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def goodNodes(self, root: TreeNode) -> int:
      result=[0]
      self.helper(root,root.val,result)
      return result[0]

    def helper(self,cur,great,result):
      if cur:
        if cur.val>=great:
          result[0]+=1
          great=cur.val
        self.helper(cur.left,great,result)
        self.helper(cur.right,great,result)